### PR TITLE
New event for slack creation2

### PR DESCRIPTION
**Pull request by Buzzglo**
### Description
to give gluon context of the user who create the team the memberId of the actioning user is added to the api call for creating the team slack channel.

### Essential Checks:

* [x] Have you added tests where necessary?
* [x] Have you added metric gathering code where necessary (in `EventHandlers` and `CommandHandlers`)?
* [x] Have you added new commands to the displayable Help list?
* [x] Have you updated any necessary documentation?

### CODE REVIEW

File: src/gluon/services/gluon/TeamService.ts [hunk 1]
```diff
@@ -114,9 +114,9 @@ export class TeamService {
         });
     }
 
-    public async addSlackDetailsToTeam(teamId: string, slackDetails: any): Promise<any> {
+    public async addSlackDetailsToTeam(teamId: string, actionedByMemberId,  slackDetails: any): Promise<any> {
```
Comment (kbristow) [inline src/gluon/services/gluon/TeamService.ts]: This should not be a new parameter. The slackDetails parameter should contain the createdBy field. So if anything the slackDetails parameter could be renamed to something more like slackUpdateDetails.

File: src/gluon/services/team/TeamSlackChannelService.ts [hunk 2]
```diff
@@ -55,7 +57,7 @@ export class TeamSlackChannelService {
 
         logger.info(`Updating team channel [${finalisedSlackChannelName}]: ${gluonTeamId}`);
 
-        const result = await this.gluonService.teams.addSlackDetailsToTeam(gluonTeamId, {
+        const result = await this.gluonService.teams.addSlackDetailsToTeam(gluonTeamId, memberId, {
```
Comment (kbristow) [inline src/gluon/services/team/TeamSlackChannelService.ts]: This should not be a new parameter but should enhance the data structure here already. Something like

```javascript
        const result = await this.gluonService.teams.addSlackDetailsToTeam(gluonTeamId, {
            createdBy: memberId,
            slack: {
                teamChannel: finalisedSlackChannelName,
            },
        }
```

File: src/gluon/services/gluon/TeamService.ts [hunk 3]
```diff
@@ -114,9 +114,9 @@ export class TeamService {
...
         logger.debug(`Trying to update team slack details. teamId: ${teamId}`);
-        return await this.axiosInstance.put(`${QMConfig.subatomic.gluon.baseUrl}/teams/${teamId}`, slackDetails);
+        return await this.axiosInstance.put(`${QMConfig.subatomic.gluon.baseUrl}/teams/${teamId}/members/${actionedByMemberId}`, slackDetails);
```
Comment (kbristow) [inline src/gluon/services/gluon/TeamService.ts]: As per the gluon PR please see that this URL should not change.

File: src/gluon/services/team/TeamSlackChannelService.ts [hunk 4]
```diff
@@ -47,6 +48,7 @@ export class TeamSlackChannelService {
 
     public async addSlackDetailsToGluonTeam(gluonTeamId: string,
                                             slackChannelName: string,
+                                            memberId,
```
Comment (kbristow) [inline src/gluon/services/team/TeamSlackChannelService.ts]: Please add a type here
```suggestion
                                            memberId: string,
```

File: src/gluon/services/gluon/TeamService.ts (hunk 3 above)
Comment (Buzzglo) [reply src/gluon/services/gluon/TeamService.ts]: corrected

**Pull request merged**